crates/polars-sql/src/sql_expr.rs

```diff
@@ -62,11 +62,30 @@ pub(crate) fn map_sql_polars_datatype(data_type: &SQLDataType) -> PolarsResult<D
         SQLDataType::Int2(_) => DataType::Int16,
         SQLDataType::Int4(_) => DataType::Int32,
         SQLDataType::Int8(_) => DataType::Int64,
-        SQLDataType::Interval => DataType::Duration(TimeUnit::Milliseconds),
+        SQLDataType::Interval => DataType::Duration(TimeUnit::Microseconds),
         SQLDataType::Real => DataType::Float32,
         SQLDataType::SmallInt(_) => DataType::Int16,
-        SQLDataType::Time { .. } => DataType::Time,
-        SQLDataType::Timestamp { .. } => DataType::Datetime(TimeUnit::Milliseconds, None),
+        SQLDataType::Time(_, tz) => match tz {
+            TimezoneInfo::None => DataType::Time,
+            _ => polars_bail!(ComputeError: "Time with timezone is not supported; found tz={}", tz),
+        },
+        SQLDataType::Timestamp(prec, tz) => {
+            let tu = match prec {
+                None => TimeUnit::Microseconds,
+                Some(3) => TimeUnit::Milliseconds,
+                Some(6) => TimeUnit::Microseconds,
+                Some(9) => TimeUnit::Nanoseconds,
+                Some(n) => {
+                    polars_bail!(ComputeError: "Unsupported timestamp precision; expected 3, 6 or 9, found prec={}", n)
+                },
+            };
+            match tz {
+                TimezoneInfo::None => DataType::Datetime(tu, None),
+                _ => {
+                    polars_bail!(ComputeError: "Timestamp with timezone is not (yet) supported; found tz={}", tz)
```

Comment:
idem.

Reply:
Fixed all of these (and a further `20 or so after searching for them!) 👌 
Related; I'll introduce a dedicated `PolarsSQLError` in a subsequent PR 🤔